pkgdb/src/util.cc

```diff
@@ -326,6 +326,58 @@ displayableGlobbedPath( const flox::AttrPathGlob & attrs )
 
 /* -------------------------------------------------------------------------- */
 
+
+std::string
+joinWithDelim( const std::vector<std::string> & strings,
+               const std::string &              delim )
+{
+  auto fold
+    = [&]( std::string acc, std::string elem ) { return acc + delim + elem; };
+  std::string joined = std::accumulate( std::next( strings.begin() ),
+                                        strings.end(),
+                                        strings[0],
+                                        fold );
+  return joined;
+}
+
+/* -------------------------------------------------------------------------- */
+
+void
+printLog( const nix::Verbosity & lvl, const std::string & msg )
+{
+  if ( lvl <= nix::verbosity ) { nix::logger->log( lvl, msg ); }
```

Comment:
question: why test the level?
nix' log should already do the filtering

Reply:
This was a snippet that @aakropotkin gave me. I don't know how the Nix logger actually works.

Reply:
The one in upstream `nix` back in 2.16 didn't actually work because of macro expansion, which is why I had been using this pattern.

If that's been fixed in 2.17.1 these aren't necessary - but they also don't hurt so I don't think this is blocking.

In any case it's nice to have our own logging functions in case we want to swap the back-end out.